sympy/solvers/bivariate.py

```diff
@@ -176,22 +179,85 @@ def _lambert(eq, x):
     p, den = den.as_coeff_Mul()
     e = exp(num/den)
     t = Dummy('t')
-    args = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
-
     # calculating solutions from args
-    for arg in args:
+    if p == 1:
+        t = e
+        args = [d/(a*b)*t]
+    elif domain.is_real:
+        ind_ls = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
+        args = []
+        j = -1
+        for i in ind_ls:
+            j += 1
+            if not isinstance(i,int):
+                if not i.has(I):
+                    args.append(ind_ls[j])
+            elif isinstance(i,int):
+                args.append(ind_ls[j])
+    else:
+        args = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
+    if len(args) == 0:
+        return S.EmptySet
+    # for simplicity defined two functions
+    # Lambert_Real_Braches , lambert_helper
+    def Lambert_Real_Braches():
         for k in lambert_real_branches:
-            w = LambertW(arg, k)
-            if k and not w.is_real:
+            rhs2 = -c/b + (a/d)*LambertW(arg, k)
+            if rhs2.has(I):
                 continue
-            rhs = -c/b + (a/d)*w
+            sol1 = lambert_helper(rhs2)
+        return sol1,rhs2
+
+    def lambert_helper(rhs):
+    # to avoid repetitive code
+        for xu in xusolns:
+            sol.append(xu.subs(u, rhs))
+        return sol
+
+    domainReal = domain.is_subset(S.Reals)
+    integer = Symbol('integer',integer=True)
```

Comment:
This should be a Dummy because this would potentially conflict with a symbol in the input equation

Reply:
a symbol can be replaced but Dummy can't be replaced...So it's difficult/not possible to work dummy in such cases...

Reply:
or we should assign such a symbol or any such variable that will not conflict..

Reply:
The function could return the dummy symbol along with the solution.

Reply:
In fact the best might be to return an `ImageSet`.

Reply:
> The function could return the dummy symbol along with the solution.
> In fact the best might be to return an ImageSet.

I agree and that's what is happening right now...
but in between the functions, there is a need for a replaceable variable as the dummy cant be replaced.

Reply:
A Dummy can be replaced if you use the exact same Dummy. That's why we need to return the Dummy that is used. An ImageSet is a good way of doing this because it bundles the Dummy up with the expression and the domain altogether.

Reply:
> A Dummy can be replaced if you use the exact same Dummy. 

I think it's not possible I've tried the index will be changed...
once a dummy is assigned e.g. n it turns into _n then how can we replace it ?

> An ImageSet is a good way of doing this because it bundles the Dummy up with the expression and the domain altogether.
to do this we have to consider lambda expression and after that, we can't replace it with 0 (as it is used in solve<to replace it with 0 to get numerical value >)

Reply:
> but in between the functions,

since the dummy variable will be assigned again (since a new variable is assigned for every function) in another function which will be different...

Reply:
```suggestion
    integersomething = Symbol('integersomething',integer=True)
```
or else something ... I know it's hacky but some alternative for this should be there as the  similar functionality to symbol...

Reply:
You just need to return the Dummy along with the expression. You shouldn't create symbols in different functions and expect them to be equal.

Reply:
> You just need to return the Dummy along with the expression. You shouldn't create symbols in different functions and expect them to be equal.

yep, the output is Dummy even after this intermediate variable...